src/waveform/renderers/waveformrendermark.cpp

```diff
@@ -63,6 +65,46 @@ void WaveformRenderMark::draw(QPainter* painter, QPaintEvent* /*event*/) {
     painter->restore();
 }
 
+void WaveformRenderMark::onSetTrack() {
+    slotCuesUpdated();
+
+    TrackPointer trackInfo = m_waveformRenderer->getTrackInfo();
+    if (!trackInfo) {
+        return;
+    }
+    connect(trackInfo.data(), SIGNAL(cuesUpdated(void)),
+                  this, SLOT(slotCuesUpdated(void)));
+}
+
+void WaveformRenderMark::slotCuesUpdated() {
+    TrackPointer trackInfo = m_waveformRenderer->getTrackInfo();
+    if (!trackInfo){
+        return;
+    }
+
+    QList<CuePointer> loadedCues = trackInfo->getCuePoints();
+
+    // This assumes no two cues can have the same hotcue assigned.
+    QList<CuePointer>::iterator it = loadedCues.begin();
+    while (it != loadedCues.end()) {
+        int hotCue = (*it)->getHotCue();
+        if (hotCue == -1) {
+            ++it;
+            continue;
+        }
+
+        QString newLabel = (*it)->getLabel();
+        QColor newColor = (*it)->getColor();
+        WaveformMark& mark = m_marks.getHotCueMark(hotCue);
```

Comment:
As I already mentioned using mutable references in this way is confusing and bad coding style.